packages/flutter/lib/src/widgets/scroll_configuration.dart

```diff
@@ -132,6 +135,26 @@ class ScrollBehavior {
   /// impossible to select text in scrollable containers and is not recommended.
   Set<PointerDeviceKind> get dragDevices => _kTouchLikeDeviceTypes;
 
+  /// A set of [LogicalKeyboardKey]s that, when any or all are pressed in
+  /// combination with a vertical pointer scroll event, will flip the axis of
+  /// the scroll input to be horizontal.
+  ///
+  /// This will for example, result in the input of a physical mouse wheel, to
+  /// apply to a [ScrollView] with an [Axis.horizontal] scroll direction.
+  ///
+  /// If other keys exclusive of this set are pressed during a scroll event, in
+  /// conjunction with keys from this set, the scroll input will not be affected.
+  ///
+  /// On MacOs, [LogicalKeyboardKey.shift] will always flip the input axes,
```

Comment:
Always?  Or never? Below it seems like it won't ever flip it, since the platform already handles it.

Reply:
Hm, I definitely want to be clear here. Shift will always flip the axis on Mac, because Mac does it. The framework does not handle the flipping.

Reply:
this might be clearer if the subject of the sentence was the piece of code doing the flipping rather than the key on which that piece of code triggered.

Reply:
I am not sure I follow, but I tweaked it. let me know if it works better.

Reply:
what i meant was saying either "the macOS device drivers flip..." or "the flutter embedder or macOS flips..." or "the flutter framework when it detects being on macOS flips...".

The shift key itself doesn't ever do any flipping of the input axis. It's just an input into a number of other logical units that do.